src/rules/selector-class-pattern/__tests__/index.js

```diff
@@ -31,3 +31,45 @@ function basicAZTests(tr) {
 
 testRule(/^[A-Z]+$/, basicAZTests)
 testRule("^[A-Z]+$", basicAZTests)
+
+function nestedAZTestsDefault(tr) {
+  warningFreeBasics(tr)
+
+  tr.ok(".AB { }")
+
+  tr.ok(".A { &__B { }}")
```

Comment:
Shouldn't this one actually fail against a `[A-Z]` regex, because it includes `_`?